lib/libc/minimal/source/bsearch.c

```diff
@@ -0,0 +1,46 @@
+/*bsearch.c - binary search routine*/
+
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <bsearch.h>
+
+/**
+ * @brief Generic implementation of Binary Search
+ *
+ * @param key	pointer to first element to search
+ * @param array	pointer to item being searched for
+ * @param count	number of elements
+ * @param size	size of each element
+ * @param cmp	pointer to comparison function
+ *
+ * @return	pointer to key if present in the array, or NULL otherwise
```

Comment:
Is 'bsearch' a part of libc?

If not it should not be placed in lib/libc/

Reply:
https://en.wikipedia.org/wiki/C_standard_library

Reply:
I see that it is a part of libc.

As was said earlier, the prototype should be put in stdlib.h.

The statement about this causing a footprint overhead in stdlib is incorrect. I don't know why Linux doesn't put it in stdlib.h. But for Zephyr it should be put into stdlib.h.

Reply:
Moved bsearch to stdlib.